framework/src/modules/chain/chain.js

```diff
@@ -114,673 +89,89 @@ module.exports = class Chain {
-
-						let privateKey;
-						let certificate;
-						let https;
-						let https_io;
-
-						if (scope.config.api.ssl && scope.config.api.ssl.enabled) {
-							privateKey = fs.readFileSync(scope.config.api.ssl.options.key);
-							certificate = fs.readFileSync(scope.config.api.ssl.options.cert);
-
-							https = require('https').createServer(
-								{
-									key: privateKey,
-									cert: certificate,
-									ciphers:
-										'ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA256:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!SRP:!CAMELLIA',
-								},
-								app
-							);
-
-							https_io = require('socket.io')(https);
-						}
-
-						cb(null, {
-							express,
-							app,
-							server,
-							io,
-							https,
-							https_io,
-						});
-					},
-				],
+		try {
+			// Cache
+			this.logger.debug('Initiating cache...');
+			const cache = createCacheComponent(cacheConfig, this.logger);
 
-				sequence: [
-					'logger',
-					/**
-					 * Description of the function.
-					 *
-					 * @func sequence[1]
-					 * @memberof! app
-					 * @param {Object} scope
-					 * @param {function} cb - Callback function
-					 * @todo Add description for the function and its params
-					 */
-					function(scope, cb) {
-						const sequence = new Sequence({
-							onWarning(current) {
-								scope.logger.warn('Main queue', current);
-							},
-						});
-						cb(null, sequence);
-					},
-				],
+			// Storage
+			this.logger.debug('Initiating storage...');
+			const storage = createStorageComponent(storageConfig, dbLogger);
 
-				balancesSequence: [
-					'logger',
-					/**
-					 * Description of the function.
-					 *
-					 * @func balancesSequence[1]
-					 * @memberof! app
-					 * @param {Object} scope
-					 * @param {function} cb - Callback function
-					 * @todo Add description for the function and its params
-					 */
-					function(scope, cb) {
-						const sequence = new Sequence({
-							onWarning(current) {
-								scope.logger.warn('Balance queue', current);
-							},
-						});
-						cb(null, sequence);
-					},
-				],
+			if (!this.options.config) {
+				throw Error('Failed to assign nethash from genesis block');
+			}
 
-				swagger: [
-					'components',
-					'modules',
-					'logger',
-					/**
-					 * Description of the function.
-					 *
-					 * @func swagger[4]
-					 * @memberof! app
-					 * @param {Object} scope
-					 * @param {function} cb - Callback function
-					 * @todo Add description for the function and its params
-					 */
-					function(scope, cb) {
-						httpApi.bootstrapSwagger(
-							scope.network.app,
-							scope.config,
-							scope.logger,
-							scope,
-							cb
-						);
+			const self = this;
```

Comment:
prefer to remove `self`

Reply:
It is required because of reference in the sequence callback.